include/sharg/parser.hpp

```diff
@@ -824,7 +853,7 @@ class parser
             }
             else
```

Comment:
This should throw invalid subcommand. If we arrive here, we know that it's not an option, so it can only be a misspell subcommand.
Remove the catch/rethrow when calling parse.